### PR TITLE
Changed the modal, nomodal, openlogin type references as per v6 release along with the migration guide

### PR DESCRIPTION
**Pull request by ihsraham**
* Changed the modal, nomodal type references as per v6 release
* Changed Openlogin adapter snippets to include privkey provider
* Added a section for Migration Guides at the root level of our docs with a basic level intro on the landing page.
* Added a sub-section as `Migrating to V6 from V5` and added content for the migration guide.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -34,16 +34,9 @@
...
-    "javascript",
-    "javascriptreact",
-    "typescript",
-    "typescriptreact"
-  ],
+  "eslint.validate": ["markdown", "mdx", "javascript", "javascriptreact", "typescript", "typescriptreact"],
   "[mdx]": {
-    "editor.defaultFormatter": "rvest.vs-code-prettier-eslint"
+    "editor.defaultFormatter": "esbenp.prettier-vscode"
   },
```
Comment (yashovardhan) [inline .vscode/settings.json:40]: I don't think these changes are needed

File: docs/sdk/pnp/web/adapters/openlogin.mdx [hunk 2]
```diff
@@ -76,21 +76,24 @@ This OpenloginAdapter constructor takes an object with `OpenloginAdapterOptions`
...
 <TabItem value="table">
 
-| Parameter          | type                                    |
-| ------------------ | --------------------------------------- |
-| `chainConfig?`     | [CustomChainConfig](#customchainconfig) |
-| `adapterSettings?` | [OpenLoginOptions](#openloginoptions)   |
-| `loginSettings?`   | [LoginSettings](#loginsettings)         |
+| Parameter             | type                                    |
+| --------------------- | --------------------------------------- |
+| `chainConfig?`        | [CustomChainConfig](#customchainconfig) |
```
[the remark this answers is not in the document]
Comment (yashovardhan) [reply docs/sdk/pnp/web/adapters/openlogin.mdx]: fixed

File: .vscode/settings.json (hunk 1 above)
Comment (yashovardhan) [reply .vscode/settings.json]: fixed

**Review by shahbaz17 — APPROVED** [review]
LGTM

**Pull request merged**